### PR TITLE
:construction: Added a screen to show a searchable list of clients filtered by village they were registered.

### PR DESCRIPTION
**Pull request by gosso22**

### CODE REVIEW

**Review by issyzac — COMMENTED** [review]
LGTM

File: app/src/main/java/org/smartregister/addo/activity/FamilyProfileActivity.java [hunk 1]
```diff
@@ -140,6 +141,13 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 // TODO Add Menu Item here
 
                 break;
+            case android.R.id.home:
+                //onBackPressed();
+                Intent intent = NavUtils.getParentActivityIntent(this);
+                assert intent != null;
```
Comment (allan-on) [inline app/src/main/java/org/smartregister/addo/activity/FamilyProfileActivity.java:147]: Not sure why the `assert` is used here? Is it an alternative to an` if...else`?

**Review by allan-on — COMMENTED** [review]
Great work @gosso22 ! 
I've left some comments mostly on readability but other than that the code looks functionally great and can be merged as is if pressed for time.

File: app/src/main/java/org/smartregister/addo/activity/FamilyProfileActivity.java (hunk 1 above)
Comment (gosso22) [reply app/src/main/java/org/smartregister/addo/activity/FamilyProfileActivity.java:147]: Not sure as well but I guess it was an IDE suggestion that was accidentally introduced

File: app/src/main/res/layout/adapter_adolescent_register_list_row.xml [hunk 2]
```diff
@@ -0,0 +1,115 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/register_columns"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:background="@color/white">
+
+    <LinearLayout
```
[the remark this answers is not in the document]
Comment (gosso22) [reply app/src/main/res/layout/adapter_adolescent_register_list_row.xml:8]: Yes, that is right

File: app/src/main/res/layout/addo_village_client_list_row.xml [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+            <org.smartregister.view.customcontrols.CustomFontTextView
+                android:id="@+id/text_view_address_gender"
+                style="@style/village_client_list_gender"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:paddingStart="0dp"
+                android:paddingEnd="0dp"
+                android:paddingTop="7dp"
+                android:text="Kitomondo . Male"/>
```
[the remark this answers is not in the document]
Comment (gosso22) [reply app/src/main/res/layout/addo_village_client_list_row.xml:87]: Yes these are placeholder texts

**Pull request merged**